scikit_mol/transformers.py

```diff
@@ -244,6 +244,81 @@ def _mol2fp(self, mol):
                                                                                     nBitsPerEntry=self.nBitsPerEntry
                                                                                     )
 
+class MHFingerprintTransformer(FpsTransformer):
+    # https://jcheminf.biomedcentral.com/articles/10.1186/s13321-018-0321-8
+    def __init__(self, radius:int=3, rings:bool=True, isomeric:bool=False, kekulize:bool=False,
+                 min_radius:int=1, n_permutations:int=2048, seed:int=42, parallel: Union[bool, int] = False,):
+        """Transforms the RDKit mol into the MinHash fingerprint (MHFP)
+
+        Args:
+            radius (int, optional): The MHFP radius. Defaults to 3.
+            rings (bool, optional): Whether or not to include rings in the shingling. Defaults to True.
+            isomeric (bool, optional): Whether the isomeric SMILES to be considered. Defaults to False.
+            kekulize (bool, optional): Whether or not to kekulize the extracted SMILES. Defaults to False.
+            min_radius (int, optional): The minimum radius that is used to extract n-gram. Defaults to 1.
+            n_permutations (int, optional): The number of permutations used for hashing. Defaults to 0, 
+            this is effectively the length of the FP
+            seed (int, optional): The value used to seed numpy.random. Defaults to 0.
+        """
+        super().__init__(parallel = parallel)
+        self.radius = radius
+        self.rings = rings
+        self.isomeric = isomeric
+        self.kekulize = kekulize
+        self.min_radius = min_radius
+        self._n_permutations = n_permutations
```

Comment:
It's not generally permitted to have properties and dynamic code in the __init__ function of the scikit-learn transformer. It also confused me when I read the init statement that you have some user settable properties in the call to __init__, that are then marked as private.
However, here it works (e.g. with cloning and get_params and set_params, as you have coded the properties with getters and setters as well. Do you do it this way here to avoid creating the encoder twice? If so, and you want to keep it this way, maybe add a comment why the private properties are manipulated in this way.

Reply:
Actually not sure if we (and I) have also done it this way elsewhere in the code.

Reply:
> It's not generally permitted to have properties and dynamic code in the **init** function of the scikit-learn transformer. It also confused me when I read the init statement that you have some user settable properties in the call to **init**, that are then marked as private.

Do you mean this piece:
`        self._n_permutations = n_permutations
        self._seed = seed
`
We have the names "n_permutations" and "seed" exposed via setter/getter, therefore we need to have a different name for the class field (otherwise there are recurrent calls).

Reply:
Yes, that one, I understand it now